rfc/development-cycle.md

```diff
@@ -0,0 +1,180 @@
+
+![](https://gist.githubusercontent.com/saulshanabrook/6d92df6e1872a5560674e097efd4abf3/raw/1e4f533196d24e9db9955dc8c0ba487e0633edc7/codepen----gitgraph-js-playground%2520(2).svg)
+
+## Division of labor
+
+There are many moving parts to the proposal. I thought it would be helpful to lay out what would be the responsibility of the different parties, including the bots.
+
+The "Bots" section is also a roadmap of what would need to be implemented to move forward on this.
+
+### Pull request author
+
+**Semver Label**: Make sure pull request has appropriate semver label.
+You can set it by either adding a commit with `semver:patch`, `semever:minor`, or `semver:major`, or setting the label in GitHub.
+
+**Changelog entry**: Add a new markdown or RST file in `docs/source/changes/new`. Use the current data and some rough title for the filename, so it is unique. If you want it to be under a subheader, make a subdirectory for that. Some common ones are `Bug fixes`, `For developers`, `User-facing changes`. If your change is small, just make it a single bullet in a list. 
+
+
+**Milestone**: Don't touch this, it will be set automatically.
+
+**Branch base**: Normally don't touch this. However, if this PR is not meant for the latest release, but instead only for a backport to an older release, set the branch base to correspond to the branch you want to merge it against. For example if you have a fix only for the next 1.2.x release, but the latest semver release is 2.1.2, then use this label and set the base branch to `1.2.x`.
+
+
+### Pull request merger
+
+**Semver Label**: Verify the correct semver label is applied to the change.
+
+**Changelog entry**: Verify that any changes add or change an existing changelog entry.
+
+**Review backports**: After you merge it, review all the backport PRs that were created and merge them if appropriate. Close any that are not relevent. Also for any backports the bot failed to make automatically, make them manually if they are necessary.
+
+### Release Manager
+
+To make a release go to {as yet unspecified page we will build} and you will see a list of all the open branches. Each branch will show you the number of commits since the last release on that branch. Also on each branch will be buttons for each of the releases you can make from that branch. Any branch which has a release ongoing (currently running) will show that. You can also delete any version branch from this UI where the last commit was a final version.
```

Comment:
I think the result of this action should be a PR somewhere that a limited subset of us can merge.  That way we defer access control to Github.

Reply:
Oooh I like that a lot! So the idea would be, on merge, an action would do a new release and push back the resulting commits?

Reply:
Ayup

Reply:
It might not have GH token rights to push directly back to jupyterlab if it is a different repo (would need to check).   In that case it could open a PR.  That PR would have to be labeled as a release in order to trigger a new set of tags.